### DISCUSSION TITLE
Setup connection of feedwater preheater with drain inlet

### DISCUSSION THREAD

**Discussion by ngom52** [oemof/tespy#487] [Q&A]
Hi everyone,

I'm very new to TESPY (literally only found out about it a few hours ago), and I'm wondering what is the proper way to set up connections for a feedwater preheater **with a drain inlet** , such as in the diagram below:

![image](https://github.com/oemof/tespy/assets/83634162/743e3b9b-2cb8-4ed1-b408-587925164b43)

I've looked into the example / tutorial [here](https://tespy.readthedocs.io/en/latest/tutorials/pygmo_optimization.html), but the feedwater preheaters are treated as condensers (no drain inlet), such as in the diagram below:

![image](https://github.com/oemof/tespy/assets/83634162/df8894ca-9799-475a-bf18-16708992cf86)

Thanks in advance for your help!

**Comment by fwitte** [comment]
Hi @ngom52,

you can add a merge upstream of the heat exchanger, would that be sufficient? Or would the equations you want to model be different from merging the two streams (extraction steam and drain) before entering the heat exchanger?

Best

**Reply by PabloMBarral** [reply]
Drain inlet should be split in sat steam and condensate. The first goes to the main inlet while the second bypasses the heater. Of course, this would be only benefitial when modelling heat transfer. As for energy and exergy it's the same.

I'm assuming no subcooling from the main outlet, though.

**Reply by fwitte** [reply]
Okay, so then you can model that by using the droplet separator upstream of the heat exchanger and merge the steam part to the extraction steam and the condensate part to the outlet of the heat exchanger.

You can then also define that component group as a subsystem and reuse it if it is frequently required.

If you want to do it in a single component, it is a bit more challenging. You'd need to inherit from the heat exchanger class and modify/add some equations plus change some parts for the internal branching system of tespy (used for mass flow and fluid composition propagation + variable space reduction).

**Reply by PabloMBarral** [reply]
Typically these preheaters have some configuration as show in this sketch:

![image](https://github.com/oemof/tespy/assets/45795487/df71e51c-a2cf-4fde-84e6-959d5b4aa129)

The simpler way to physically add a new connection is somehow drilling the shell to weld another connection. This way, the heater can be modelled as two heaters (desuperheating zone + subcooling zone) and a condensing heater. In the latter, the entire process consists in steam condensing over a tube bundle, so a flashed condensate incoming from another preheater isn't really a perturbation or something that changes radically the nature of the heat transfer (as it would occur if you do not separate the zones).

Perhaps defining a new class (as explained in the GT discussion posted some time ago in this same forum) could be helpful. 

But with regard to energy, splitting the model won´t add relevant information. The same does not occur with exergy efficiency (not entirely sure, I should do the math). Of course, the required area for heat transfer, as the DELTA T changes and so does the transfer mode (less convective desuperheating, more condensing).

Hope this helps, and feel free to correct any misinterpretation of mine.

**Reply by ngom52** [reply]
Hi everyone,

Thanks for all the suggestions, they are all very much appreciated. I happened to find an example using feedwater heaters with drain inlets [here](https://github.com/fwitte/SEGS_exergy). I happen to have some familiarity & a trial license for Ebsilon, and the Ebsilon model in that link actually does make use of that feedwater heater configuration.

In that example, the TESPy model is represented by a condenser and a drain + extraction steam merge upstream of of the feedwater heater, so I'll try that out first.

Thanks again for all the suggestions!

**Reply by fwitte** [reply]
You're welcome, and thanks to @PabloMBarral for sharing your knowledge here! :)
